### PR TITLE
SONAR-3821 Drop any Maven dependency

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

**Comment by simonbrandhof** [comment]
As discussed this PR must not be merged as long as Java plugin is coupled to Maven libraries.

File: plugins/sonar-batch-maven-plugin/src/main/java/org/sonar/plugins/batch/maven/MavenProjectBuilder.java [hunk 1]
```diff
@@ -51,7 +51,8 @@ public void build(Context context) {
   private void setMavenProjectIfApplicable(ProjectDefinition definition) {
     if (mavenSession != null) {
       String moduleKey = definition.getKey();
-      for (MavenProject mavenModule : (List<MavenProject>) mavenSession.getSortedProjects()) {
+      for (MavenProject mavenModule : (List<MavenProject>) mavenSession.getProjects()) {
+        // FIXME assumption that moduleKey was not modified by user and follow convention <groupId>:<artifactId>
```
Comment (SonarTech) [inline plugins/sonar-batch-maven-plugin/src/main/java/org/sonar/plugins/batch/maven/MavenProjectBuilder.java]: :new: MAJORTake the required action to fix the issue indicated by this comment.[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)

File: plugins/sonar-batch-maven-plugin/src/main/java/org/sonar/plugins/batch/maven/MavenProjectBuilder.java [hunk 2]
```diff
@@ -51,7 +51,8 @@ public void build(Context context) {
   private void setMavenProjectIfApplicable(ProjectDefinition definition) {
     if (mavenSession != null) {
       String moduleKey = definition.getKey();
-      for (MavenProject mavenModule : (List<MavenProject>) mavenSession.getSortedProjects()) {
+      for (MavenProject mavenModule : (List<MavenProject>) mavenSession.getProjects()) {
```
Comment (SonarTech) [inline plugins/sonar-batch-maven-plugin/src/main/java/org/sonar/plugins/batch/maven/MavenProjectBuilder.java]: :new: MINORRemove this unnecessary cast to "List".[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1905)

File: sonar-batch/src/main/java/org/sonar/batch/scan/filesystem/ProjectFileSystemAdapter.java [hunk 3]
```diff
@@ -50,20 +43,14 @@
...
   private final PathResolver pathResolver = new PathResolver();
-  private final MavenProject pom;
 
-  public ProjectFileSystemAdapter(DefaultModuleFileSystem target, Project project, @Nullable MavenProject pom) {
+  public ProjectFileSystemAdapter(DefaultModuleFileSystem target, Project project) {
     this.target = target;
-    this.pom = pom;
 
     // previously MavenProjectBuilder was responsible for creation of ProjectFileSystem
     project.setFileSystem(this);
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/scan/filesystem/ProjectFileSystemAdapter.java]: MINORMethod 'Project.setFileSystem(...)' is deprecated.[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 new issues:
* 6 critical
* 1 major
* 1 minor

Details:
* [Remove this unnecessary cast to "List".](https://github.com/SonarSource/sonarqube/blob/76121590fd420b300f58084c1e6d3254920cf992/plugins/sonar-batch-maven-plugin/src/main/java/org/sonar/plugins/batch/maven/MavenProjectBuilder.java#L54)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1905)
* [Take the required action to fix the issue indicated by this comment.](https://github.com/SonarSource/sonarqube/blob/76121590fd420b300f58084c1e6d3254920cf992/plugins/sonar-batch-maven-plugin/src/main/java/org/sonar/plugins/batch/maven/MavenProjectBuilder.java#L55)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)
* [Method 'public java.lang.String getPackaging()' has been removed](https://github.com/SonarSource/sonarqube/blob/76121590fd420b300f58084c1e6d3254920cf992/sonar-plugin-api/src/main/java/org/sonar/api/resources/Project.java)[](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* [Method 'public org.apache.commons.configuration.Configuration getConfiguration()' has been removed](https://github.com/SonarSource/sonarqube/blob/76121590fd420b300f58084c1e6d3254920cf992/sonar-plugin-api/src/main/java/org/sonar/api/resources/Project.java)[](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* [Method 'public org.sonar.api.resources.Project setPom(org.apache.maven.project.MavenProject)' has been removed](https://github.com/SonarSource/sonarqube/blob/76121590fd420b300f58084c1e6d3254920cf992/sonar-plugin-api/src/main/java/org/sonar/api/resources/Project.java)[](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* [Method 'public org.sonar.api.resources.Project setPackaging(java.lang.String)' has been removed](https://github.com/SonarSource/sonarqube/blob/76121590fd420b300f58084c1e6d3254920cf992/sonar-plugin-api/src/main/java/org/sonar/api/resources/Project.java)[](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* [Method 'public java.lang.String[] getTestExclusionPatterns()' has been removed](https://github.com/SonarSource/sonarqube/blob/76121590fd420b300f58084c1e6d3254920cf992/sonar-plugin-api/src/main/java/org/sonar/api/resources/Project.java)[](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* [Method 'public java.lang.String[] getExclusionPatterns()' has been removed](https://github.com/SonarSource/sonarqube/blob/76121590fd420b300f58084c1e6d3254920cf992/sonar-plugin-api/src/main/java/org/sonar/api/resources/Project.java)[](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 new issues:
* 6 critical
* 1 major
* 1 minor

Details:
* Take the required action to fix the issue indicated by this comment. (org.codehaus.sonar.plugins:sonar-batch-maven-plugin:src/main/java/org/sonar/plugins/batch/maven/MavenProjectBuilder.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)
* Remove this unnecessary cast to "List". (org.codehaus.sonar.plugins:sonar-batch-maven-plugin:src/main/java/org/sonar/plugins/batch/maven/MavenProjectBuilder.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1905)
* Method 'public java.lang.Object getProperty(java.lang.String)' has been removed (org.codehaus.sonar:sonar-plugin-api:src/main/java/org/sonar/api/resources/Project.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* Method 'public org.apache.commons.configuration.Configuration getConfiguration()' has been removed (org.codehaus.sonar:sonar-plugin-api:src/main/java/org/sonar/api/resources/Project.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* Method 'public org.sonar.api.resources.Project setConfiguration(org.apache.commons.configuration.Configuration)' has been removed (org.codehaus.sonar:sonar-plugin-api:src/main/java/org/sonar/api/resources/Project.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* Method 'public java.lang.String[] getTestExclusionPatterns()' has been removed (org.codehaus.sonar:sonar-plugin-api:src/main/java/org/sonar/api/resources/Project.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* Method 'public org.apache.maven.project.MavenProject getPom()' has been removed (org.codehaus.sonar:sonar-plugin-api:src/main/java/org/sonar/api/resources/Project.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)
* Method 'public java.lang.String getArtifactId()' has been removed (org.codehaus.sonar:sonar-plugin-api:src/main/java/org/sonar/api/resources/Project.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=clirr%3Aclirr-api-break)

**Pull request merged**